### ISSUE TITLE
Enable logs RBAC enforcement by default

### ISSUE THREAD

**Issue by crenshaw-dev** [argoproj/argo-cd#10551]
# Summary

In 2.4 we introduced new RBAC for logs. In 2.5, we should enable that enforcement by default as [promised in the 2.3 -> 2.4 upgrade guide](https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/upgrading/2.3-2.4.md#enable-logs-rbac-enforcement).

# Motivation

We said we'd do it. :-)

# Proposal

Set the flag to true by default, and add a note to the 2.4 -> 2.5 upgrade notes.

**Comment by crenshaw-dev** [comment]
I think we should save this for 3.0.

Project-scoped roles can currently only hold RBAC for the `applications` resource. So this change would break logs for anyone who relies on a Project-scoped role implicitly granting logs access via `applications, get`.

**Comment by niqdev** [comment]
I know that this is not enabled by default in 2.x, but after upgrading to argo-helm `5.8.7`, I had to explicitly add `p, role:admin, logs, get, */*, allow` for the admin user to re-enable the logs.

Shouldn't [this](https://github.com/argoproj/argo-cd/blob/master/assets/builtin-policy.csv#L16) be added also to the admin by default?

**Comment by agaudreault** [comment]
Project-scoped roles should support `logs` resource before this can be enabled by default IMO. It makes more sense to allow/deny logs within an application or project instead of globally. It also aligns with the principle of least privilege that is currently configurable in the Project resource.

**Comment by reggie-k** [comment]
Actually, projects scoped roles already support logs.

**Comment by reggie-k** [comment]
> # Proposal
> Set the flag to true by default, and add a note to the 2.4 -> 2.5 upgrade notes.

@crenshaw-dev 
Won't we better just remove the flag? Otherwise it leaves an option to be overridden with false.

**Comment by agaudreault** [comment]
@reggie-k I think the goal of 3.0 is to have a way to opt out of each breaking change. We should flag it as deprecated in 3.0, and remove it for 4.0 IMO

**Comment by reggie-k** [comment]
@agaudreault I see your point.
However, leaving the fallback option to disable the rbac enforcement preserves the state of logs rbac as a second-class rbac citizen.
Opting out if this default (in other words, giving logs access to everyone) can be done in another way, by explicitly providing those permissions, even without the flag. I believe that for other breaking changes in this release there is only way to opt-out of, so we are just changing defaults. For this one, we have an alternative.
WDYT?

**Comment by agaudreault** [comment]
@reggie-k that makes total sense 😃

**Comment by reggie-k** [comment]
After further research, if we delete the flag, there would be no easy option to opt-out.
An explicit permission grant would be needed per each AppProject that previously had an implicit grant.

I am reverting the change that deleted the switch and simply flipping the default to `true`.

**Comment by reggie-k** [comment]
@agaudreault I agree, it can be done on the ConfigMap globally, but not with a single setting. It would require a configuration line per each AppProject having `get` or `*` on `applications`.
I definitely prefer removing all the code supporting the switch and making this secure-by-default (and gladly did before realizing opting out is not a single setting flip) - just not sure where do we draw the line of "easy opt-out".

Do you feel it is reasonable to drop the switch and just clearly document how to give permissions for each AppProject globally for a quicker migration?

**Comment by reggie-k** [comment]
After further discussion with @agaudreault, there IS a quick migration procedure, which can be performed with adding just 1-2 lines in the ConfigMap. So we drop support for the switch and add clearer documentation.
